src/lib/form/composeAsyncValidators/types.ts

```diff
@@ -0,0 +1,6 @@
+type ReturnValue = string | undefined
+
+export type Validator<Value, FormValues = object> = (
```

Comment:
I think it would be more clear if this was name something like `FormFieldValidator`. We already have a type from `ui` called `Validator`.